src/Tizen.Network.Stc/Tizen.Network.Stc/StcManagerImpl.cs

```diff
@@ -0,0 +1,218 @@
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "GetStats() failed , Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+
+            return task.Task;
+        }
+
+        internal Task<IEnumerable<NetworkStatistics>> GetAllStatisticsAsync(StcRule rule)
+        {
+            if (rule._disposed)
+            {
+                throw new ArgumentException("Invalid StcRule object (Object may have been disposed or released).");
+            }
+
+            TaskCompletionSource<IEnumerable<NetworkStatistics>> task = new TaskCompletionSource<IEnumerable<NetworkStatistics>>();
+            IntPtr id;
+
+            lock (_getAllStatsCb_map)
+            {
+                id = (IntPtr)_requestAllId++;
+                _getAllStatsCb_map[id] = (int result, IntPtr infoList, IntPtr key) =>
+                {
+                    if(result != (int)StcError.None)
+                    {
+                        Log.Error(Globals.LogTag, "GetAllStats failed, Error - " + (StcError)result);
+                        task.SetException(new InvalidOperationException("Error occurs during GetAllStats(), " + (StcError)result));
+                    }
+
+                    List<NetworkStatistics> statsList = new List<NetworkStatistics>();
+
+                    Interop.Stc.StatsInfoCallback foreachAllStatsCb = (int resultTemp, Interop.Stc.SafeStatsHandle info, IntPtr userDataTemp) =>
+                    {
+                        if(resultTemp != (int)StcError.None)
+                        {
+                            Log.Error(Globals.LogTag, "ForeachAllStats failed, Error - " + (StcError)resultTemp);
+                            task.SetException(new InvalidOperationException("Error occurs during ForeachAllStats(), " + (StcError)resultTemp));
+                        }
+
+                        Interop.Stc.SafeStatsHandle cloned;
+                        int retValue = Interop.Stc.Info.StatsClone(info, out cloned);
+                        if (retValue != (int)StcError.None)
+                        {
+                            Log.Error(Globals.LogTag, "StatsClone() failed , Error - " + (StcError)retValue);
+                            task.SetException(new InvalidOperationException("Error occurs during StatsClone(), " + (StcError)retValue));
+                        }
+
+                        statsList.Add(new NetworkStatistics(cloned));
+                        return CallbackRet.Continue;
+                    };
+
+                    int retTemp = Interop.Stc.ForeachAllStats(infoList, foreachAllStatsCb, IntPtr.Zero);
+                    if(retTemp != (int)StcError.None)
+                    {
+                        Log.Error(Globals.LogTag, "foreachAllStatus() failed , Error - " + (StcError)retTemp);
+                        StcErrorFactory.ThrowStcException(retTemp);
```

Comment:
you should not throw exception in here
use task.SetException

Reply:
Done.